IPs/RSKIP_tbd.md

```diff
@@ -0,0 +1,282 @@
+5. Activation phase: new Powpeg address activates, two-way peg starts operating from this new address.
+6. Migration phase: Funds from the retiring Powpeg wallet are moved to the active Powpeg.
+7. Post migration phase: Retiring Powpeg information is removed from the Bridge. New elections are re-enabled.
+
+### Commit new Powpeg
+
+The commitment of a new Powpeg composition remains almost unchanged except for a subtle need.
+Given the election could be reverted (see [here](#election-rollback)) the Bridge will have to store the last retired Powpeg script in a backup until the validation phase finishes.
+
+**Important**: This needs to take place before the current powpeg p2sh script is written in the storage.
+
+Suggested implementation:
+```
+KEY: "SVP_LAST_RETIRED_P2SHSCRIPT_BKP"
+VALUE: value of lastRetiredFedP2SHScript before commitment of new Powpeg
+```
+
+
+### New phase - Validation
+
+If this RSKIP is implemented a new phase will be added, in between the _pre activation_ and _activation_ phases, the _validation_ phase. This phase will, in a sense, replace the _preactivation_ phase.
+
+During a certain number of blocks a new protocol will be started, Spendability Validation Protocol (SVP).
+If this protocol succeeds, the Bridge will progress to the next phase (_activation_).
+If this protocol fails the election will be rollbacked.
+
+### Spendability Validation Protocol (SVP)
+
+The Spendability Validation Protocol is a new process in which the soon-to-be-retired Powpeg will send funds to the newly elected Powpeg, for it to send them back, proving that it can truly hold and spend funds, before proceeding with the activation and migration of the funds in the wallet.
+This protocol involves three parts:
+- New Powpeg funding: The new Powpeg won't have funds to create the proof so the current Powpeg will create a UTXO covering the expenses.
+- Proof transaction creation: Once the new Powpeg receives the funds, it will create a UTXO to the current Powpeg to proof that it can spend funds.
+- Proof transaction verification: The current Powpeg will receive the proof, validate it was created by the new Powpeg and proceed.
+
+#### New Powpeg funding
+
+Once the SVP starts, the Bridge will send the minimum required satoshis to the new Powpeg address. It should be enough for the new Powpeg to be able to send back a transaction. The size of the Proof transaction will vary depending on the size of the new Powpeg input script.
+
+The process consists of the following 3 steps.
+
+##### Funding transaction creation
+
+As soon as a new Powpeg has been committed, the Bridge will create a SPV funding transaction for the new Powpeg.
+
+This transaction is similar to a peg-out where the recipient is the new Powpeg Bitcoin address, and the amount to receive is calculated based on the minimum amount of satoshis required to spend a UTXO by the new Powpeg.
+
+** :warning: TODO: HACER CALCULOS REALES**
+
+Possible calculation:
```

Comment:
The virtual size of a transaction is: (baseSize * 3 + totalSize) / 4
The calculation of the baseSize will depend on if the new powpeg is legacy / segwit compatible / segwit native.
The baseSize includes: the input -that includes the scriptSig- size, the output size and a few more bytes (version, locktime, etc).
The totalSize also includes the "witness".
If it is legacy, there is no witness and the scriptSig contains all the signatures and the redeemScript. Therefore, the baseSize and the totalSize are the same.
If it is segwit compatible, the scriptSig is a fixed 32-bytes value and the witness contains the signatures and the redeemScript.
If it is segwit native, the scriptSig is empty and the witness contains the signatures and the redeemScript.

Reply:
Since we will spend just one input, we should add 36 bytes from the utxo itself (fundTxHash and the index); and since we will have just one output, we should add approximately 36 bytes too. From those few more bytes, approximately 10 more.

`bytesFromInputAndOutput = 36 + 36 + 10 = 82 approx.`
So for all addresses types, `baseSize = bytesFromInputAndOutput + bytesFromScriptSig`
And `bytesFromEmergencyPublicKeys = 4 * 33 = 132`

```
legacy =>
_bytesFromScriptSig_ = (NS * 73) from signatures + (NS * 34 (standard public keys) + 13 (opcodes from multisig script) + bytesFromEmergencyPublicKeys) from redeemScript
= _NS*73 + NS*34 + 13 + 132_
=> _legacyBaseSize_ = 82 + NS*73 + NS*34 + 13 + 132 = _227 + NS * 73 + NS * 34_
_witness_ = _0_
=> legacyTotalSize = 0 + legacyBaseSize = legacyBaseSize
=> **legacyVirtualSize** = (legacyBaseSize*3 + legacyTotalSize) / 4 = (legacyTotalSize*3 + legacyTotalSize) / 4
= legacyTotalSize = **227 + NS * 73 + NS * 34**
```

```
segwit compatible =>
_bytesFromScriptSig_ = _32_
=> _segwitCompatibleBaseSize_ = 82 + 32 = _114_
_witness_ = (NS * 73) from signatures + (NS * 34 (standard public keys) + NS * 3 - 2 (opcodes from custom multisig script) + bytesFromEmergencyPublicKeys) from redeemScript
= _NS*73 + NS*34 + NS*3-2 + 132_
=> _segwitCompatibleTotalSize_ = segwitCompatibleBaseSize + witness = 114 + (NS*73 + NS*34 + NS*3-2 + 132)
= _246 + NS*73 + NS*34 + NS*3-2_
=> **segwitCompatibleVirtualSize** = (segwitCompatibleBaseSize*3 + segwitCompatibleTotalSize) / 4
= (114*3 + 246 + NS*73 + NS*34 + NS*3-2) / 4 = **(588 + NS*73 + NS*34 + NS*3) / 4**
```

```
segwit native => 
is the same as segwitCompatible besides the fact that with native the scriptSig is empty, so it adds 0 bytes instead of 32.
bytesFromScriptSig = 0
=> _segwitNativeBaseSize_ = _82_
=> _segwitNativeTotalSize_ = _214 + NS*73 + NS*34 + NS*3-2_
=> **segwitNativeVirtualSize** = (segwitNativeBaseSize*3 + segwitCompatibleTotalSize) / 4
= (82*3 + 214 + NS*73 + NS*34 + NS*3-2) / 4 = **(460 + NS*73 + NS*34 + NS*3) / 4**
```


**disclaimer: we should discuss if this tx includes flyover or not**

Reply:
So the satoshis to be sent for fees is
`virtualSize * FEE_PER_BYTE`